src/commands/initProjectForVSCode/verifyVSCodeConfig.ts

```diff
@@ -138,37 +140,136 @@ async function verifyJavaDeployConfigIsValid(projectLanguage: ProjectLanguage |
         if (result === DialogResponses.dontWarnAgain) {
+            actionContext.properties.verifyConfigResult = 'dontWarnAgain';
             await updateWorkspaceSetting(settingKey, false, fsPath);
         } else {
+            actionContext.properties.verifyConfigResult = 'update';
             return true;
         }
     }
 
     return false;
 }
 
+async function verifyTargetFramework(projectLanguage: ProjectLanguage, workspacePath: string, projectPath: string, actionContext: IActionContext): Promise<void> {
+    const settingKey: string = 'showTargetFrameworkWarning';
+    if (getFuncExtensionSetting<boolean>(settingKey)) {
+
+        const projFileName: string | undefined = projectLanguage === ProjectLanguage.CSharp ? await tryGetCsprojFile(projectPath) : await tryGetFsprojFile(projectPath);
+        if (projFileName) {
+
+            const targetFramework: string | undefined = await tryGetTargetFramework(path.join(projectPath, projFileName));
+            if (targetFramework) {
+
+                const tasksResult: IVerifyFrameworkResult | undefined = verifyTasksFramework(workspacePath, targetFramework);
+                const settingsResult: IVerifyFrameworkResult | undefined = verifySettingsFramework(workspacePath, targetFramework);
+
+                const mismatchTargetFramework: string | undefined = (tasksResult && tasksResult.mismatchTargetFramework) || (settingsResult && settingsResult.mismatchTargetFramework);
+                if (mismatchTargetFramework) {
+                    actionContext.properties.verifyConfigPrompt = 'updateTargetFramework';
+
+                    // This won't handle the case where there are multiple different target frameworks, but it's good enough for the message
+                    const message: string = localize('mismatchTargetFramework', 'The targetFramework "{0}" in your project file does not match the targetFramework "{1}" in your VS Code config.', targetFramework, mismatchTargetFramework);
+                    const update: vscode.MessageItem = { title: localize('updateTargetFramework', 'Update VS Code config') };
+
+                    const result: vscode.MessageItem = await ext.ui.showWarningMessage(message, update, DialogResponses.dontWarnAgain);
+                    if (result === DialogResponses.dontWarnAgain) {
+                        actionContext.properties.verifyConfigResult = 'dontWarnAgain';
+                        await updateWorkspaceSetting(settingKey, false, workspacePath);
+                    } else if (result === update) {
+                        actionContext.properties.verifyConfigResult = 'update';
+                        if (tasksResult) {
+                            await tasksResult.update();
+                        }
+
+                        if (settingsResult) {
+                            await settingsResult.update();
+                        }
+                    }
+                }
+            }
+        }
```

Comment:
Could be useful to add a telemetry event here to see how many times this prompt gets ignored.